File: sdks/conformance/python/conformance.py
"""Live conformance: drive the GENERATED Python control-plane client against a
running beaterd and verify typed request/response shapes match the API.

Proves API-shape == SDK-shape for Python. Run via run.sh.
"""

import os
import sys

import beater_client
from beater_client import ApiClient, Configuration
from beater_client.api.health_api import HealthApi
from beater_client.api.traces_api import TracesApi
from beater_client.api.datasets_api import DatasetsApi
from beater_client.models.create_dataset_request import CreateDatasetRequest

BASE = os.environ["BEATER_BASE_URL"].rstrip("/")
TENANT = os.environ.get("BEATER_TENANT", "demo")
PROJECT = os.environ.get("BEATER_PROJECT", "demo")


def main() -> int:
    cfg = Configuration(host=BASE)
    with ApiClient(cfg) as api:
        # 1. health -> typed response
        health = HealthApi(api).health()
        assert getattr(health, "ok") is True, f"health.ok != True: {health}"
        print(f"  health: ok={health.ok}")

        # 2. create dataset -> typed request body + typed response (shape parity)
        created = DatasetsApi(api).create_dataset(
            TENANT, PROJECT, CreateDatasetRequest(name="conformance-py")
        )
        print(f"  createDataset -> {type(created).__name__}")

        # 3. list traces -> typed page response
        page = TracesApi(api).list_traces(TENANT)
        items = getattr(page, "items", None)
        assert items is not None, f"traces.list page missing 'items': {page}"
        print(f"  traces.list -> {type(page).__name__} items={len(items)}")

    print("PASS: python generated client round-trips against live API")
    return 0


if __name__ == "__main__":
    try:
        sys.exit(main())
    except Exception as exc:  # noqa: BLE001
        print(f"FAIL: {type(exc).__name__}: {exc}", file=sys.stderr)
        sys.exit(1)

File: sdks/conformance/ruby/conformance.rb
# frozen_string_literal: true

# Live conformance: drive the GENERATED Ruby control-plane client against a
# running beaterd and verify typed request/response shapes match the API.
#
# Proves API-shape == SDK-shape for Ruby. Run via run.sh.

require "uri"
require "beater_client"

base = ENV.fetch("BEATER_BASE_URL").sub(%r{/+\z}, "")
tenant = ENV.fetch("BEATER_TENANT", "demo")
project = ENV.fetch("BEATER_PROJECT", "demo")
uri = URI.parse(base)

BeaterClient.configure do |config|
  config.scheme = uri.scheme
  config.host = "#{uri.host}:#{uri.port}"
  config.server_index = nil
end

begin
  # 1. health -> typed response
  health = BeaterClient::HealthApi.new.health
  raise "health.ok != true: #{health.inspect}" unless health.ok == true

  puts "  health: ok=#{health.ok}"

  # 2. create dataset -> typed request body + typed response (shape parity)
  req = BeaterClient::CreateDatasetRequest.new(name: "conformance-ruby")
  created = BeaterClient::DatasetsApi.new.create_dataset(tenant, project, req)
  puts "  createDataset -> #{created.class.name.split('::').last}"

  # 3. list traces -> typed page response
  page = BeaterClient::TracesApi.new.list_traces(tenant)
  raise "traces.list page missing 'items': #{page.inspect}" if page.items.nil?

  puts "  traces.list -> #{page.class.name.split('::').last} items=#{page.items.length}"

  puts "PASS: ruby generated client round-trips against live API"
rescue StandardError => e
  warn "FAIL: #{e.class}: #{e.message}"
  exit 1
end
